helios-testing/src/main/java/com/spotify/helios/testing/TemporaryJobs.java

```diff
@@ -499,35 +248,44 @@ static Builder builder(final Map<String, String> env) {
   }
 
   static Builder builder(final String profile, final Map<String, String> env) {
-    return builder(profile, env, HeliosClient.newBuilder());
+    return new Builder(profile, HeliosConfig.loadConfig("helios-testing"), env);
   }
 
-  static Builder builder(final String profile, final Map<String, String> env,
-                         final HeliosClient.Builder clientBuilder) {
-    return new Builder(profile, HeliosConfig.loadConfig("helios-testing"), env, clientBuilder);
+  @Override
+  public void close() {
+    final List<AssertionError> errors = new ArrayList<>();
+
+    for (final TemporaryJob job : jobs) {
+      // Undeploying a job doesn't guarantee the container will be stopped immediately.
+      // Luckily TaskRunner tells docker to wait 120 seconds after stopping to kill the container.
+      // So containers that don't immediately stop **should** only stay around for at most 120
+      // seconds.
+      errors.addAll(undeploy(job.job(), job.hosts()));
+    }
+
+    for (final AssertionError error : errors) {
+      log.error(error.getMessage());
+    }
```

Comment:
> If TemporaryJobs now managed HeliosSoloDeployment, is it intended or accidental that it doesn't close the HSD instance in TemporaryJobs.close()?

Intended, if you ask me. I think it makes sense to decouple the lifetime of the `HeliosDeployment` from the lifetime of the `TemporaryJobs` instance.

For example, you might want to reuse `HeliosDeployment` across multiple tests, but clean up jobs after each of them. I think it's fair to make the user responsible for closing the `HeliosDeployment`.